pkg/model/clickhouse/connection.go

```diff
@@ -63,7 +69,7 @@ func (c *Connection) SetLog(l log.Announcer) *Connection {
 // connect performs connect
 func (c *Connection) connect(ctx context.Context) {
 	c.l.V(2).Info("Establishing connection: %s", c.params.GetDSNWithHiddenCredentials())
-	dbConnection, err := databasesql.Open("clickhouse", c.params.GetDSN())
+	dbConnection, err := databasesql.Open("clickhouse", c.params.GetDSN()+"?tls_config=tls-settings")
```

Comment:
Just thinking aloud - may be these params should be not added to `GetDSN()`, but produced by `GetDSN()`?

Reply:
Especially taking into account, that DSN is already aware of the `scheme` (be it http/https)

Reply:
I think I ended going this way because of this unit was importing the go driver here, so the crypto init/registration had to also happen here. The specific registration key is currently hard coded on unit initialization and wanted to confine the hardcoded key to the scope of this unit.  
I can make it a constant constant ("tls-settings") in the ConnectionParam unit.